nginx/ng/install.sls

```diff
@@ -55,9 +55,9 @@ nginx_yum_repo:
     - name: nginx
     - humanname: nginx repo
     {%- if salt['grains.get']('os') == 'CentOS' %}
-    - baseurl: 'http://nginx.org/packages/centos/$releasever/$basearch/'
+    - baseurl: 'http://nginx.org/packages/centos/{{ nginx.lookup.rh_os_releasever }}/$basearch/'
```

Comment:
CentOS is always going to be correct, you only need the lookup in the else case (which is the code path Amazon Linux  takes)